source/empty-object.d.ts

```diff
@@ -0,0 +1,39 @@
+declare const emptyObjectSymbol: unique symbol;
+
+/**
+Represents a strictly empty object. In other words, this type accepts `{}` only.
+
+@example
+```
+import type {EmptyObject} from 'type-fest';
+
+let foo: EmptyObject = {};
+
+foo = {}; // Pass
+
+foo = []; // Error
+foo = 42; // Error
+foo = null; // Error
+foo.bar = {}; // Error
+```
+
```

Comment:
Maybe we can also mention at the end here that `Record<string, never>`, `Record<keyof any, never>`, and `Record<never, never>` do not work and link to https://github.com/sindresorhus/type-fest/issues/395 I'm sure some users will ask why not use one of these solutions instead.